src/utils/globalScript.ts

```diff
@@ -7,7 +7,7 @@ import { initThemeChangeEvent } from "./theme";
  */
 export default function (): void {
   if (isBrowser()) {
-    initThemeChangeEvent();
+    document.addEventListener("DOMContentLoaded", () => initThemeChangeEvent());
```

Comment:
Do you need to check if the content has already been loaded and therefore the event already fired?